crypto/keyring/keyring.go

```diff
@@ -582,29 +575,31 @@ func (ks keystore) NewAccount(name string, mnemonic string, bip39Passphrase stri
 	// if found
 	address := sdk.AccAddress(privKey.PubKey().Address())
 	if _, err := ks.KeyByAddress(address); err == nil {
-		return nil, fmt.Errorf("account with address %s already exists in keyring, delete the key first if you want to recreate it", address)
+		return nil, errors.New("duplicated address created")
 	}
 
-	return ks.writeLocalKey(name, privKey, algo.Name())
+	return ks.writeLocalKey(name, privKey)
 }
 
 func (ks keystore) isSupportedSigningAlgo(algo SignatureAlgo) bool {
 	return ks.options.SupportedAlgos.Contains(algo)
 }
 
-func (ks keystore) key(infoKey string) (Info, error) {
-	bs, err := ks.db.Get(infoKey)
+func (ks keystore) Key(uid string) (*Record, error) {
+	if _, err := ks.migrate(uid); err != nil {
```

Comment:
`migrate(uid)` can return a `*Record` or `Item` - so we don't need the code below to re query the key.